### PR TITLE
Add/Edit variables to BMI

### LINKED ISSUE #111: Model is not finalizing
Hello @BSchilperoort 
I noticed that the model is not finalized with an error message if the model is running for longer periods (e.g. 20 days). I did several tests with different sites but always same problem. I actually can't figure out where is the problem, so would appreciate your help. I attached the model here in case you could test from your side.

[NL-Loo_2024-11-13-1159.zip](https://github.com/user-attachments/files/17735617/NL-Loo_2024-11-13-1159.zip)

Screenshot 2024-11-13 170345

### LINKED ISSUE #114: the EVAP variable is always zero in BMI
The `Èvap` variable is always zero through BMI, although it has a value. I checked that if we delete the `all_timesteps = True` in this [line](https://github.com/EcoExtreML/STEMMUS_SCOPE_Processing/blob/33adb9e19ecdedfb8446aec5a848374d416d7cfa/PyStemmusScope/bmi/variable_reference.py#L38C1-L38C28), the `Èvap` variable will get its value in BMI

### LINKED ISSUE #115: Add precipitation, infiltration and transpiration variables to BMI
Three variables are added to BMI
1. Precipitation ([in those lines](https://github.com/EcoExtreML/STEMMUS_SCOPE_Processing/blob/cd28ad6fe5e528af7b73f1eae345257cfce682a4/PyStemmusScope/bmi/variable_reference.py#L108C1-L116C7))
2. Infiltration ([in those lines](https://github.com/EcoExtreML/STEMMUS_SCOPE_Processing/blob/cd28ad6fe5e528af7b73f1eae345257cfce682a4/PyStemmusScope/bmi/variable_reference.py#L117C1-L126C7))
3. Transpiration ([in those lines](https://github.com/EcoExtreML/STEMMUS_SCOPE_Processing/blob/cd28ad6fe5e528af7b73f1eae345257cfce682a4/PyStemmusScope/bmi/variable_reference.py#L40C1-L48C7)). The `Trap` variable is still not in the [BMI list](https://github.com/EcoExtreML/STEMMUS_SCOPE/blob/b9b4f8231955589d3394ec5148950467eeaa4ec3/src/STEMMUS_SCOPE_exe.m#L21C1-L36C30) in the source code, but will be added in the next source code release

### PR DESCRIPTION
**Pull request by MostafaGomaa93**
closes #114 
closes #115
closes #111

### CODE REVIEW

File: PyStemmusScope/bmi/variable_reference.py [hunk 1]
```diff
@@ -35,7 +35,16 @@ class BmiVariable:
         units="cm s-1",
         grid=0,
         keys=["EVAP"],
-        all_timesteps=True,
+        # all_timesteps=True,
```
Comment (BSchilperoort) [inline PyStemmusScope/bmi/variable_reference.py:38]:
```suggestion
```

**Comment by SarahAlidoost** [comment]
@MostafaGomaa93 and @BSchilperoort the failed [markdown-link-check](https://github.com/EcoExtreML/STEMMUS_SCOPE_Processing/actions/runs/12375979294/job/34542032766?pr=116) might be related to [this issue](https://github.com/gaurav-nelson/github-action-markdown-link-check/issues/185). I submitted issue #119.

**Review by SarahAlidoost — APPROVED** [review]
@MostafaGomaa93 thanks for adding new variables to bmi and fixing some issues.  The changes look good and the notebook `bmi_demo` works fine. The documentation about `SleepDuration` will be added in #118 where I am updating docs.

File: PyStemmusScope/bmi/variable_reference.py [hunk 2]
```diff
@@ -28,15 +28,43 @@ class BmiVariable:
...
+        name="evaporation_total",
+        dtype="float64",
+        input=False,
+        output=True,
+        units="cm s-1",
+        grid=0,
+        keys=["Evap"],
+    ),
+    BmiVariable(
+        name="transpiration_total",
```
Comment (BSchilperoort) [inline PyStemmusScope/bmi/variable_reference.py:60]: I can understand the "evaporation_total" name, as transpiration is separate (and you can have soil evaporation separately as well) but why does transpiration have `_total` in the name?

You cannot get `transpiration_understory` and `transpiration_canopy` right?

**Review by BSchilperoort — APPROVED** [review]
Looks largely OK. Just have a comment on a variable name.

Can you also list the newly available variables in the changelog?

File: PyStemmusScope/bmi/variable_reference.py (hunk 2 above)
Comment (MostafaGomaa93) [reply PyStemmusScope/bmi/variable_reference.py:60]: I actually agree with you, I was trying to give a similar name to `evaporation_total`.  However, I don't get why it is called `evaporation_total`. What STEMMUS-SCOPE calculates is `soil_evaporation` and not total evaporation. Physically, there is still one evaporation component that is missing (called `canopy interception`), but it is not simulated by STEMMUS_SCOPE yet. 

So maybe we should change the names 
`evaporation_total` to `evaporation`
`transpitation_total` to `transpiration`

Comment (BSchilperoort) [reply PyStemmusScope/bmi/variable_reference.py:60]: > What STEMMUS-SCOPE calculates is soil_evaporation and not total evaporation. 

Then the evaporation variable should be named `soil_evaporation`!

**Pull request merged**